test/integration/registration/integration_suite_test.go

```diff
@@ -224,32 +249,106 @@ var _ = ginkgo.BeforeSuite(func() {
 		}()
 	}
 
-	startHub(hubOption)
+	switch authDriver {
+	case helpers.CSRAuthType:
+		hubOption = hub.NewHubManagerOptions()
+		hubOption.EnabledRegistrationDrivers = []string{helpers.CSRAuthType}
+		hubOption.ClusterAutoApprovalUsers = []string{util.AutoApprovalBootstrapUser}
 
-	// start a proxy server
-	proxyCertData, proxyKeyData, err := authn.SignServerCert("proxyserver", 24*time.Hour)
-	gomega.Expect(err).ToNot(gomega.HaveOccurred())
-	proxyServer := util.NewProxyServer(proxyCertData, proxyKeyData)
-	ctx, stopProxy = context.WithCancel(context.Background())
-	err = proxyServer.Start(ctx, 5*time.Second)
-	gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		registerDriverOption = func(_ string) *factory.Options {
+			return factory.NewOptions()
+		}
 
-	httpProxyURL = proxyServer.HTTPProxyURL
-	httpsProxyURL = proxyServer.HTTPSProxyURL
+		startHub(hubOption)
+
+		// start a proxy server
+		proxyCertData, proxyKeyData, err := authn.SignServerCert("proxyserver", 24*time.Hour)
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		proxyServer := util.NewProxyServer(proxyCertData, proxyKeyData)
+		ctx, stopProxy = context.WithCancel(context.Background())
+		err = proxyServer.Start(ctx, 5*time.Second)
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+		httpProxyURL = proxyServer.HTTPProxyURL
+		httpsProxyURL = proxyServer.HTTPSProxyURL
+
+		// create bootstrap hub kubeconfig with http/https proxy settings
+		bootstrapKubeConfigHTTPProxyFile = path.Join(util.TestDir, "bootstrap-http-proxy", "kubeconfig")
+		err = authn.CreateBootstrapKubeConfigWithProxy(bootstrapKubeConfigHTTPProxyFile, serverCertFile, securePort, httpProxyURL, nil)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		bootstrapKubeConfigHTTPSProxyFile = path.Join(util.TestDir, "bootstrap-https-proxy", "kubeconfig")
+		err = authn.CreateBootstrapKubeConfigWithProxy(bootstrapKubeConfigHTTPSProxyFile, serverCertFile, securePort, httpsProxyURL, proxyCertData)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	case helpers.GRPCCAuthType:
+		tempDir, err := os.MkdirTemp("", "registration-test")
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		gomega.Expect(tempDir).ToNot(gomega.BeEmpty())
+
+		bootstrapConfigFileName := path.Join(tempDir, "grpcconfig")
+		_, gRPCServerOptions, caKeyFile, err := util.CreateGRPCConfigs(bootstrapConfigFileName)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		hubOption = hub.NewHubManagerOptions()
+		hubOption.EnabledRegistrationDrivers = []string{helpers.GRPCCAuthType}
+		hubOption.GRPCCAFile = gRPCServerOptions.ClientCAFile
+		hubOption.GRPCCAKeyFile = caKeyFile
+		// TODO test auto approval
+		hubOption.ClusterAutoApprovalUsers = []string{util.AutoApprovalBootstrapUser}
+
+		registerDriverOption = func(hubConfigDir string) *factory.Options {
```

Comment:
this is confusing, why a func with specifying hubConfigDir will result in the type to be grpc? I think we should have a NewOption() func in test that build option based on type.